docs/AUTH_PROTOCOL.md

```diff
@@ -0,0 +1,393 @@
+# Proposal for alternative design for JSON messages
+
+This is a proposal for a new JSON-based authentication protocol for GDM (and, eventually, the systemd upstream replacement for GDM and other desktop-specific display managers). The authenticator 
+
+This protocol is event driven. Basically, the two parties (the authenticator, which I'll be calling the "broker". And the frontend, which I'll be calling "GDM") will pass events back and forth at each other as things happen. Events have a basic structure:
+
+```js
+{
+    "event": [some event],
```

Comment:
what is an event here? an enum-like string?

Reply:
Yep (or its related ID) if varlink allows alternatively to use one or another)